provider/marathon/marathon_test.go

```diff
@@ -934,440 +700,565 @@ func TestMarathonGetProtocol(t *testing.T) {
+		groupAsSubDomain bool
 	}{
-		{"/test", "test", providerNoGroups},
-		{"/test", "test", providerGroups},
-		{"/a/b/c/d", "d.c.b.a", providerGroups},
-		{"/b/a/d/c", "c.d.a.b", providerGroups},
-		{"/d/c/b/a", "a.b.c.d", providerGroups},
-		{"/c/d/a/b", "b.a.d.c", providerGroups},
-		{"/a/b/c/d", "a-b-c-d", providerNoGroups},
-		{"/b/a/d/c", "b-a-d-c", providerNoGroups},
-		{"/d/c/b/a", "d-c-b-a", providerNoGroups},
-		{"/c/d/a/b", "c-d-a-b", providerNoGroups},
+		{"/test", "test", false},
+		{"/test", "test", true},
+		{"/a/b/c/d", "d.c.b.a", true},
+		{"/b/a/d/c", "c.d.a.b", true},
+		{"/d/c/b/a", "a.b.c.d", true},
+		{"/c/d/a/b", "b.a.d.c", true},
+		{"/a/b/c/d", "a-b-c-d", false},
+		{"/b/a/d/c", "b-a-d-c", false},
+		{"/d/c/b/a", "d-c-b-a", false},
+		{"/c/d/a/b", "c-d-a-b", false},
 	}
 
-	for _, a := range apps {
-		actual := a.provider.getSubDomain(a.path)
-
-		if actual != a.expected {
-			t.Errorf("expected %q, got %q", a.expected, actual)
-		}
+	for _, c := range cases {
+		c := c
+		t.Run(fmt.Sprintf("path=%s,group=%t", c.path, c.groupAsSubDomain), func(t *testing.T) {
+			t.Parallel()
+			provider := &Provider{GroupsAsSubDomains: c.groupAsSubDomain}
+			actual := provider.getSubDomain(c.path)
+			if actual != c.expected {
+				t.Errorf("actual %q, expected %q", actual, c.expected)
+			}
+		})
 	}
 }
 
 func TestMarathonHasHealthCheckLabels(t *testing.T) {
-	tests := []struct {
-		desc  string
-		value *string
-		want  bool
+	cases := []struct {
+		desc     string
+		value    *string
+		expected bool
 	}{
 		{
-			desc:  "label missing",
-			value: nil,
-			want:  false,
+			desc:     "label missing",
+			value:    nil,
+			expected: false,
 		},
 		{
-			desc:  "empty path",
-			value: stringp(""),
-			want:  false,
+			desc:     "empty path",
+			value:    stringp(""),
+			expected: false,
 		},
 		{
-			desc:  "non-empty path",
-			value: stringp("/path"),
-			want:  true,
+			desc:     "non-empty path",
+			value:    stringp("/path"),
+			expected: true,
 		},
 	}
 
-	for _, test := range tests {
-		test := test
-		t.Run(test.desc, func(t *testing.T) {
+	for _, c := range cases {
+		c := c
+		t.Run(c.desc, func(t *testing.T) {
 			t.Parallel()
-			app := marathon.Application{
-				Labels: &map[string]string{},
-			}
-			if test.value != nil {
-				app.AddLabel(types.LabelBackendHealthcheckPath, *test.value)
+			app := application()
+			if c.value != nil {
+				app.AddLabel(types.LabelBackendHealthcheckPath, *c.value)
 			}
-			prov := &Provider{}
-			got := prov.hasHealthCheckLabels(app)
-			if got != test.want {
-				t.Errorf("got %t, want %t", got, test.want)
+			provider := &Provider{}
+			actual := provider.hasHealthCheckLabels(app)
+			if actual != c.expected {
+				t.Errorf("actual %t, expected %t", actual, c.expected)
 			}
 		})
 	}
 }
 
 func TestMarathonGetHealthCheckPath(t *testing.T) {
-	tests := []struct {
-		desc  string
-		value *string
-		want  string
+	cases := []struct {
+		desc     string
+		value    *string
```

Comment:
You're right: There's still one test left where we want to distinguish set/unset/omitted, but not this one.

Simplified.